src/main/java/filter/FilterException.java

```diff
@@ -0,0 +1,8 @@
+package filter;
+
+@SuppressWarnings("serial")
```

Comment:
I guess we could document this as well, as being a general filter-related error. Maybe it should be abstract, so we ensure that something that's actually thrown has a concrete type/reason for happening?